server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/utils/search/MailboxFilter.scala

```diff
@@ -49,4 +53,57 @@ object MailboxFilter {
     List(InMailboxFilter, NotInMailboxFilter).foldLeft(multiMailboxQueryBuilder)((builder, filter) => filter.toQuery(builder, request))
       .build()
   }
+
+
+  sealed trait QueryFilter {
+    def toQuery(builder: SearchQuery.Builder, request: EmailQueryRequest): SearchQuery.Builder
+  }
+
+  object QueryFilter {
+    def buildQuery(request: EmailQueryRequest): SearchQuery.Builder = {
+      List(ReceivedBefore, ReceivedAfter, HasKeyWord, NotKeyWord).foldLeft(new SearchQuery.Builder())((builder, filter) => filter.toQuery(builder, request))
+    }
+  }
+
+  case object ReceivedBefore extends QueryFilter {
+    override def toQuery(builder: SearchQuery.Builder, request: EmailQueryRequest): SearchQuery.Builder =  request.filter.flatMap(_.before) match {
+      case Some(before) =>
+        val strictlyBefore = new InternalDateCriterion(new DateOperator(DateComparator.BEFORE, Date.from(before.asUTC.toInstant), DateResolution.Second))
+        val sameDate = new InternalDateCriterion(new DateOperator(DateComparator.ON, Date.from(before.asUTC.toInstant), DateResolution.Second))
+        new SearchQuery.Builder()
```

Comment:
You need to carry other the existing builder overwize previous criteria are lost?

(here and in other QueryFilter?)